### PR TITLE
Improve Site Init Page

### LINKED ISSUE #6420: Site init page shows in dark theme if OS dark theme is enabled
In macOS if you have the systemwide dark theme enabled, the site init page shows up in a dark theme because the default theme behavior is to match the system's behavior. This is a problem because the site init page is not designed to look good in the dark theme.

Fix: make the site init page look good with a dark theme

image

### LINKED ISSUE #8715: Icons not centered in text
image

Reported by @eseliger

### PR DESCRIPTION
**Pull request by felixfbecker**
- Fixes #6420 
- Fixes #8715 
- removes a lot of CSS cruft

image

image

image

image

### CODE REVIEW

File: shared/src/global-styles/icons.scss [hunk 1]
```diff
@@ -8,9 +8,7 @@
 // Formats an icon so that it appears inline in the text
 // with height matching the text
 .icon-inline {
-    // SVGs already include padding to account for text baseline,
-    // so align with the bottom of the text
-    vertical-align: bottom;
+    vertical-align: top;
```
Comment (eseliger) [inline shared/src/global-styles/icons.scss]: could you add a screenshot of the search syntax info sheet where the issue was originally reported on? :)

File: web/src/auth/SignUpForm.tsx [hunk 2]
```diff
@@ -85,17 +91,23 @@ export class SignUpForm extends React.Component<SignUpFormProps, SignUpFormState
...
+                        <div className="form-check">
+                            <label className="form-check-label">
+                                <input
+                                    className="form-check-input"
+                                    type="checkbox"
+                                    onChange={this.onRequestTrialFieldChange}
+                                />
+                                Try Sourcegraph Enterprise free for 30 days{' '}
+                                {/* eslint-disable-next-line react/jsx-no-target-blank */}
+                                <a target="_blank" href="https://about.sourcegraph.com/pricing">
```
Comment (eseliger) [inline web/src/auth/SignUpForm.tsx]:
```suggestion
                                <a target="_blank"  rel="noopener noreferrer" href="https://about.sourcegraph.com/pricing">
```

Comment (felixfbecker) [reply web/src/auth/SignUpForm.tsx]: This is our own site so we cab trust it and I imagine it's useful to track which site referred to the pricing page

File: shared/src/global-styles/icons.scss (hunk 1 above)
Comment (felixfbecker) [reply shared/src/global-styles/icons.scss]: Ah, I was wondering where that screenshot came from. Turns out that icon actually applied a bottom margin for seemingly no reason...

File: web/src/auth/SignUpForm.tsx (hunk 2 above)
Comment (eseliger) [reply web/src/auth/SignUpForm.tsx]: Fair, lets remove the noreferrer 
To the noopener:

> When you open another page using target="_blank", the other page may run on the same process as your page, unless Site Isolation is enabled. If the other page is running a lot of JavaScript, your page's performance may also suffer.

Source: https://developers.google.com/web/tools/lighthouse/audits/noopener

**Pull request merged**